File: php/src/Matrix/Sudoku.php
<?php

namespace IbrahimMH\LeetCode\Solution\Matrix;


class Sudoku{



    public function isValid(array $borad){

        $rows = array_fill(0,9,[]);
        $columns = array_fill(0,9,[]);
        $suboxs = array_fill(0,9,[]);

        for($i=0;$i < 9; $i++){
               for($j=$i;$j<=$i;$j++){

                $cell = $borad[$i][$j];

                if($cell == '.') continue;

                $boxIndex = ($i%3) * 3 + ($j%3);

                # checker
                if(in_array($cell,$rows) ||
                   in_array($cell,$columns) || 
                   in_array($cell,$suboxs[$boxIndex]))
                   {
                    return false;
                }
                echo "\n($i,$j)\n";
                # add to our logs
                $rows[$i][] = $cell;
                $columns[$j][] = $cell;
                $suboxs[$boxIndex][] = $cell;
               }
        }
        return true;
    }

}

File: python/matrix/sudoku.py




def is_valid_sudoku(borad: list) ->bool:

   rows = [set() for _ in range(9) ]
   columns = [set() for _ in range(9) ]
   suboxs = [set() for _ in range(9) ]

   for row in range(9):
      
      for column in range(9):
         
        cell = borad[row][column]

        if cell == '.': 
            continue
        
        subox_index = ( row // 3 ) * 3 + ( column // 3 ) 
        
        if ( cell in rows[row] or
             cell in columns[column] or
             cell in suboxs[subox_index] ):

            return False 

        rows[row].add(cell)
        columns[column].add(cell)
        suboxs[subox_index].add(cell)

   return True     